docker-compose-dev.yml

```diff
@@ -1,18 +1,57 @@
 version: '3'
 
 services:
-    hapi:
-        build: './hapi_fhir'
+    # Database service for broker
+    #
+    broker-db:
+        build: ./broker_db
+        environment:
+            # Variables read from local .env file
+            - POSTGRES_DB=${BROKER_DB_NAME}
+            - POSTGRES_USER=${BROKER_DB_USERNAME}
+            - POSTGRES_PASSWORD=${BROKER_DB_PASSWORD}
+        volumes:
+            #- broker_db_config:/etc/postgresql
+            #- broker_db_logs:/var/log/postgresql
+            - broker_db_data:/var/lib/postgresql/data
         ports:
-            - "8080:8080"
+            - "6432:5432"
+
+    # Database for HAPI FHIR service
+    #
+    hapi-db:
+        image: postgres
+        environment:
+            # Variables read from local .env file
+            - POSTGRES_DB=${HAPI_DB_NAME}
+            - POSTGRES_USER=${HAPI_DB_USERNAME}
+            - POSTGRES_PASSWORD=${HAPI_DB_PASSWORD}
         volumes:
-            - ./hapi_fhir/data/hapi:/data/hapi
-    db:
-        build: './postgres_db'
+            #- hapi_db_config:/etc/postgresql
+            #- hapi_db_logs:/var/log/postgresql
+            - hapi_db_data:/var/lib/postgresql/data
         ports:
             - "5432:5432"
+
+    # HAPI FHIR service
+    #
+    hapi:
+        image: hapiproject/hapi:v5.0.0
         volumes:
-            - pgdata:/var/lib/postgresql/data/
-# named volume for persistent database
+            - ./hapi_fhir/data/hapi/hapi.properties:/etc/hapi/hapi.properties
```

Comment:
As the HAPI DB name/user/password are hard-coded in this file, having the ability to define those in the HAPI_DB_* env vars doesn't make much sense for the dev environment. Those can probably be removed and hard-coded above to match.